### PR TITLE
Fix for spark map_tuple

### PR DESCRIPTION
**Pull request by dvadym**
map_tuples was implemented as map in SparkRDDPipelineOperation. This PR fixes that and adds tests.

Along the way, the creation of SparkRDDOperations() was moved to setUpClass(cls), to match BeamOperations unittests.

### CODE REVIEW

**Comment by boyaryn** [comment]
I guess we expect `SparkRDDOperations` to be stateless. What do you think about declaring its methods class methods?

File: tests/pipeline_operations_test.py [hunk 1]
```diff
@@ -140,21 +138,19 @@ def test_filter_by_key_nonempty_public_partitions(self, distributed):
...
         self.assertTrue(set(result[1]).issubset({11, 14, 16}))
         self.assertSetEqual(set(result[2]), {22, 25})
         self.assertSetEqual(set(result[3]), {33})
 
     def test_count_per_element(self):
-        spark_operations = SparkRDDOperations()
         data = ['a', 'b', 'a']
         dist_data = SparkRDDOperationsTest.sc.parallelize(data)
-        rdd = spark_operations.count_per_element(dist_data)
+        rdd = self.ops.count_per_element(dist_data)
```
Comment (boyaryn) [inline tests/pipeline_operations_test.py:153]: Could you also `replace spark_operations` with `self.ops` in the following method `test_reduce_accumulators_per_key`?
I couldn't figure out how to make a comment to a line which wasn't added by you. :-)

**Review by boyaryn — COMMENTED** [review]
Thanks, the PR is good, only one small thing was overlooked.

File: tests/pipeline_operations_test.py (hunk 1 above)
Comment (dvadym) [reply tests/pipeline_operations_test.py:153]: Thanks, fixed. I don't know a way to comment on not-changed lines either :)

**Review by boyaryn — APPROVED** [review]

**Comment by dvadym** [comment]
> I guess we expect `SparkRDDOperations` to be stateless. What do you think about declaring its methods class methods?

Now `PipelineOperations` is stateless, but it's not necessary to be so, e.g. we might want to manage `stage_name` in `BeamOperations` (managing stage_name is pretty complicate in Beam, sometimes to avoid duplicate names (which are not allowed) scopes are introduced. So, I think it's better to keep methods `PipelineOperations` to instance methods, anyway it doesn't add any overhead (nor programming/nor performance)

**Comment by dvadym** [comment]
Thanks for the review!

**Pull request merged**